Sources/VLCMediaFileDiscoverer.h

```diff
@@ -14,10 +14,8 @@
 
 @protocol VLCMediaFileDiscovererDelegate <NSObject>
 
-@required
-- (void)mediaFilesFoundRequiringAdditionToStorageBackend:(NSArray <NSString *> *)foundFiles;
-
 @optional
+- (void)mediaFilesFoundRequiringAdditionToStorageBackend:(NSArray <NSString *> *)foundFiles;
```

Comment:
why did you move it to optional and do we still need it if it's not implemented?

Reply:
As I can see it's used in tvOS and can't say if it's really needed but I really don't want to start refactoring tvOS related code here. 

Additionally, my goal for this PR is not the remove the usage of `VLCMediaFileDiscoverer`.

For the why I removed it, as written in the commit description, I changed it to optional since the new medialibrary doesn't need a callback for `mediaFilesFoundRequiringAdditionToStorageBackend`.